### DISCUSSION TITLE
Resetting a form using field array with new changed values

### DISCUSSION THREAD

**Discussion by nlrowe** [react-hook-form/react-hook-form#2726] [Q&A] [answered]
I have some nested field arrays in a form that I do not navigate away from on a successful submit. The dirty flag is used to disable/enable the submit button and after the submit is successful, the form still shows as dirty as I assume it is comparing the new data to the initial data the form was loaded with.

To address this, I started calling the forms reset function with the form data that is passed to my submit handler. However, this has caused some odd issues that can be [reproduced here](https://codesandbox.io/s/react-hook-form-usefieldarray-forked-mdq5k?file=/src/nestedArray.js).

## Issue 1
1. Click append nested
2. Enter value into input
3. Click submit

### Result
New item disappears from the list

### Expected Result
All items remain in the form and the dirty flag is reset to false

## Issue 2
1. Click remove on item 'A'
2. Click submit

### Result
Item 'B' is replaced with item 'A'

### Expected Result
Item 'B' is the only item in the form and the dirty flag is reset to false
   
    
Is this a correct way to handle this scenario?

**Comment by bluebill1049** [comment, marked as answer]
Thanks for the detailed issue report! any chance to do a check with the latest version? I will take a look it at as well.

**Reply by nlrowe** [reply]
Looks like it has been fixed in a newer version.
